projectq/ops/_basics.py

```diff
@@ -295,23 +295,29 @@ def get_merged(self, other):
 
     def __eq__(self, other):
         """ Return True if same class and same rotation angle. """
-        tolerance = 1.e-12
-        if isinstance(other, self.__class__):
-            difference = abs(self._angle - other._angle)
```

Comment:
It won't give the wrong result. There are two cases when this `__eq__` should return true. Case 1: 
`if difference < tolerance:`
and case 2:
`if difference > 4 * math.pi - tolerance:`
both of these cases are handled correctly